service/history/workflow/mutable_state_impl.go

```diff
@@ -2463,6 +2463,18 @@ func (ms *MutableStateImpl) AddWorkflowExecutionStartedEventWithOptions(
 	); err != nil {
 		return nil, err
 	}
+
+	// Versioning Override set on StartWorkflowExecutionRequest
+	if startRequest.GetStartRequest().GetVersioningOverride() != nil {
+		metrics.WorkerDeploymentVersioningOverrideCounter.With(
+			ms.metricsHandler.WithTags(
+				metrics.NamespaceTag(ms.namespaceEntry.Name().String()),
+				metrics.VersioningBehaviorBeforeOverrideTag(enumspb.VERSIONING_BEHAVIOR_UNSPECIFIED),
+				metrics.VersioningBehaviorAfterOverrideTag(startRequest.GetStartRequest().GetVersioningOverride().GetBehavior()), //nolint:staticcheck // SA1019: worker versioning v0.31
+				metrics.VersioningOverrideOnNewWorkflowTag(prevRunID == ""),
```

Comment:
it'd be super cool if instead we could have the following values:
`_new`, `_child`, `_can`, `_retry`, `_cron`. For the last three you can look at the [initiator](https://github.com/temporalio/api/blob/350edebf546cc43c1a545fe665bb99dfbf7c194b/temporal/api/history/v1/message.proto#L52) field.

Add a `_existing` to the list to cover the case of UpdateOptions.